types/hapi/v17/test/server/server-cache-provision.ts

```diff
@@ -0,0 +1,21 @@
+//  https://github.com/hapijs/hapi/blob/master/API.md#-await-servercacheprovisionoptions
+import {Server} from "hapi";
+import * as catbox from "catbox";
+
+const server = new Server({
+    port: 8000,
+});
+server.initialize();
+server.cache.provision({engine: require('catbox-memory'), name: 'countries' });
+
+const cache:catbox.Policy = server.cache({segment: 'countries', cache: 'countries', expiresIn: 60 * 60 * 1000 });
+cache.set('norway', 'oslo', 10 * 1000, null);
+const value = cache.get('norway', null);
+
+// TODO Not working. It needs review.
```

Comment:
All the examples I am compiling (obviously) AND executing with NodeJS. TS compiler is good but some problems I can see only when I execute it. In this particular case, there is an exception in execution time. So I am not sure if the problem is my definitions, is the example or the hapijs framework.

I think the problem is with the example code. But I am not sure. I would like to confirm it with someone.

```AssertionError [ERR_ASSERTION]: Unknown cache countries
    at module.exports.internals.Core._cachePolicy (D:\Workspace\DefinitelyTyped-Rafael\node_modules\hapi\lib\core.js:520:14)
    at internals.Server.policy [as cache] (D:\Workspace\DefinitelyTyped-Rafael\node_modules\hapi\lib\server.js:513:27)
    at Object.<anonymous> (D:\Workspace\DefinitelyTyped-Rafael\types\hapi\v17\dist\server\server-cache-provision.js:10:20)
    at Module._compile (module.js:641:30)
    at Object.Module._extensions..js (module.js:652:10)
    at Module.load (module.js:560:32)
    at tryModuleLoad (module.js:503:12)
    at Function.Module._load (module.js:495:3)
    at Function.Module.runMain (module.js:682:10)
    at startup (bootstrap_node.js:191:16)
[nodemon] app crashed - waiting for file changes before starting...```

Reply:
Did you copy it from here https://github.com/hapijs/hapi/blob/master/API.md#-await-servercacheprovisionoptions ? Because one difference I see is that the example uses `await`, which we don't in our test, here...